src/main/java/com/google/googleidentity/oauth2/token/JdbcOAuth2TokenService.java

```diff
@@ -347,9 +327,46 @@ public boolean revokeByAccessToken(String accessToken) {
         || Instant.ofEpochSecond(token.get().getExpiredTime()).isBefore(Instant.now())) {
       return false;
     }
+    if (!Strings.isNullOrEmpty(token.get().getRefreshToken())) {
+      return revokeByRefreshToken(token.get().getRefreshToken());
+    }
 
-    revokeUserClientTokens(token.get().getUsername(), token.get().getClientId());
-
+    Connection conn = null;
+    PreparedStatement statement = null;
+    ResultSet result = null;
+    try {
+      conn = dataSource.getConnection();
+      conn.setAutoCommit(false);
+      String stmt = "DELETE FROM access_token WHERE access_token = ?;";
+      statement = conn.prepareStatement(stmt);
+      statement.setString(1, accessToken);
+      statement.execute();
+      conn.commit();
+    } catch (SQLException exception) {
+      log.log(Level.INFO, "Revoke token error.", exception);
```

Comment:
Shall throw out the exception.

Reply:
just throw exception?

Reply:
Yes, when we code, instead of tolerating those unexpected errors, we'd better throw the exception even that will fail the service, but it give us a chance to detect and fix it at very early stage. Otherwise if we somehow make it keep working, that will likely accumulate the dirty data and crash one day when it goes too far, and that will be very hard to debug as well. There are many articles about "dont' fail silently", you may take a look if interested in. https://google.com/search?q=don%27t+fail+silently